lib/auth/auth_with_roles.go

```diff
@@ -3400,17 +3406,45 @@ func (a *ServerWithRoles) CreateUser(ctx context.Context, user types.User) (type
 	if err := a.action(apidefaults.Namespace, types.KindUser, types.VerbCreate); err != nil {
 		return nil, trace.Wrap(err)
 	}
+
+	if a.hasBuiltinRole(types.RoleOkta) && !hasOriginOkta(user) {
+		return nil, trace.BadParameter("Okta service must supply okta origin")
+	}
+
 	created, err := a.authServer.CreateUser(ctx, user)
 	return created, trace.Wrap(err)
 }
 
+func (a *ServerWithRoles) isOktaUser(ctx context.Context, username string) error {
+	targetUser, err := a.authServer.GetUser(ctx, username, false)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+
+	if hasOriginOkta(targetUser) {
+		return nil
+	}
+
+	return trace.AccessDenied("Okta service may only update okta users")
+}
+
 // UpdateUser updates an existing user in a backend.
 // Captures the auth user who modified the user record.
 func (a *ServerWithRoles) UpdateUser(ctx context.Context, user types.User) (types.User, error) {
 	if err := a.action(apidefaults.Namespace, types.KindUser, types.VerbUpdate); err != nil {
 		return nil, trace.Wrap(err)
 	}
 
+	if a.hasBuiltinRole(types.RoleOkta) {
+		if !hasOriginOkta(user) {
```

Comment:
Similar to the other conditional recommendation, we need to verify that someone is not setting an account as Okta without having the Okta role.

Reply:
It's perfectly legitimate for a non-Okta user to Update a user with the Okta origin label. Editing via the UI, for example. We can possibly check that the user is not _adding_ origin label - but that might be getting a bit convoluted.

Reply:
I am not sure I understand the use case of manually modifying Okta users.

Would that not result in whatever manual changes were done to be overdone if a change occurred on Okta?  Or is there a sync that occurs into Okta?

> We can possibly check that the user is not adding origin label

I think we should at a minium

Reply:
There is a sync process; basically all traits with an `okta/` prefix will be updated to match the upstream Okta profile. An admin should be able to manually add roles to the user and have them preserved.

Reply:
What triggers this sync?  How do we know the sync will happen in sequence to other updates?

Reply:
Sync runs every _n_ minutes, where _n_ is currently `10`, but we may change it depending on how it interacts with Okta API rate limits.

Reply:
Since Okta is often driven by automation, I feel like there is a serious risk that a change made on Teleport could easily be undone.  If this is a necessary use case it may be better to understand what changed within the user and commit the change so that we can apply the diffs across updates.  As an MVP it would be simpler to avoid allowing Okta users to be managed outside of Okta.